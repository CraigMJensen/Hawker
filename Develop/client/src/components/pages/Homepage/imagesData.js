// Car Images Import
import blackCar from '../../../assets/images/auto/black_car.jpg';
import carCover from '../../../assets/images/auto/car_cover.jpg';
import carJack from '../../../assets/images/auto/car_jack.jpg';
import carVacuum from '../../../assets/images/auto/car_vacuum.jpg';
import carSeat from '../../../assets/images/auto/child-car-seat.jpg';
import pinkAccesories from '../../../assets/images/auto/fuzzy_pink_accesories.jpg';
import jacks from '../../../assets/images/auto/Jack-Side-by-Side.jpg';
import minivan from '../../../assets/images/auto/minivan.jpg';
import silverCar from '../../../assets/images/auto/silver_car.jpg';
import slightlyUsed from '../../../assets/images/auto/slightly_used.png';

// Clothing Images Import
import babyClothes from '../../../assets/images/clothing/baby_clothes.jpg';
import childPjs from '../../../assets/images/clothing/child_pjs.jpg';
import dressShoes from '../../../assets/images/clothing/dress_shoes.jpg';
import greenShirt from '../../../assets/images/clothing/green_shirt.png';
import mensPants from '../../../assets/images/clothing/mens_pants1.jpg';
import mensShirt from '../../../assets/images/clothing/mens_shirt1.jpg';
import pizzaShoes from '../../../assets/images/clothing/pizza_shoes.jpg';
import rainbowShirt from '../../../assets/images/clothing/rainbow_shirt.jpg';
import womensPants from '../../../assets/images/clothing/womens_pants1.jpg';
import womensShirt from '../../../assets/images/clothing/womens_shirt2.jpg';

// Household Images Import
import bigLamp from '../../../assets/images/household/big_lamp.jpeg';
import blueChair from '../../../assets/images/household/blue_chair.jpg';
import bookshelf from '../../../assets/images/household/bookshelf.jpg';
import colander from '../../../assets/images/household/colander.jpg';
import comfyChair from '../../../assets/images/household/comfy_chair.jpeg';
import couch from '../../../assets/images/household/couch.jpg';
import lamp from '../../../assets/images/household/lamp.jpg';
import standingLight from '../../../assets/images/household/standing_light.jpg';
import vacuum from '../../../assets/images/household/vacuum.jpg';
import woodenSpoons from '../../../assets/images/household/wooden_spoons.jpg';

// Outdoor Images Imports
import blackEgg from '../../../assets/images/outdoor/acapulco-black-egg-outdoor-chair.jpg';
import outdoorLights from '../../../assets/images/outdoor/outdoor_lights.jpg';
import pathLights from '../../../assets/images/outdoor/path_lights.jpg';
import patioChairSet from '../../../assets/images/outdoor/patio_chair_set.jpeg';
import patioSet from '../../../assets/images/outdoor/patio_set.jpg';
import poolChair from '../../../assets/images/outdoor/pool_chair.jpg';
import poolSet from '../../../assets/images/outdoor/pool_set.jpg';
import rotisserie from '../../../assets/images/outdoor/rotisserie.jpg';
import smoker from '../../../assets/images/outdoor/smoker.jpg';
import swingChair from '../../../assets/images/outdoor/swing_chair.jpg';

// Tech Images Imports
import pcSpeaker from '../../../assets/images/tech/computer-speaker.jpg';
import cooler from '../../../assets/images/tech/cooler.jpg';
import cryorigCooler from '../../../assets/images/tech/cryorig-cooler.jpg';
import keyboardMouse from '../../../assets/images/tech/keyboard_mouse.jpg';
import msi3080 from '../../../assets/images/tech/MSI_3080.jpg';
import redBlackGpu from '../../../assets/images/tech/red_black_gpu.jpg';
import rgbFan from '../../../assets/images/tech/rgb_fan.jpg';
import rgbKeyboard from '../../../assets/images/tech/rgb_keyboard.jpg';
import speakerSet from '../../../assets/images/tech/speaker_set.jpg';
import woodenSpeakers from '../../../assets/images/tech/wooden_speakers.jpg';

const Images = [
  {
    src: blackCar,
    alt: 'Black Car',
    title: 'Black Car'
  },
  {
    src: carCover,
    alt: 'Car Cover',
    title: 'Car Cover'
  },
  {
    src: carJack,
    alt: 'Car Jack',
    title: 'Car Jack'
  },
  {
    src: carVacuum,
    alt: 'Car Vacuum',
    title: 'Car Vacuum'
  },
  {
    src: carSeat,
    alt: 'Child Car Seat',
    title: 'Child Car Seat'
  },
  {
    src: pinkAccesories,
    alt: 'Fuzzy Pink Accesories',
    title: 'Fuzzy Pink Accesories'
  },
  {
    src: jacks,
    alt: 'Jacks',
    title: 'Jacks'
  },
  {
    src: minivan,
    alt: 'Minivan',
    title: 'Minivan'
  },
  {
    src: silverCar,
    alt: 'Silver Car',
    title: 'Silver Car'
  },
  {
    src: slightlyUsed,
    alt: 'Slightly Used Car',
    title: 'Slightly Used Car'
  },
  {
    src: babyClothes,
    alt: 'Baby Clothes',
    title: 'Baby Clothes'
  },
  {
    src: childPjs,
    alt: 'Child PJs',
    title: 'Child PJs'
  },
  {
    src: dressShoes,
    alt: 'Dress Shoes',
    title: 'Dress Shoes'
  },
  {
    src: greenShirt,
    alt: 'Green Shirt',
    title: 'Green Shirt'
  },
  {
    src: mensPants,
    alt: 'Mens Pants',
    title: 'Mens Pants'
  },
  {
    src: mensShirt,
    alt: 'Mens Shirt',
    title: 'Mens Shirt'
  },
  {
    src: pizzaShoes,
    alt: 'Pizza Shoes',
    title: 'Pizza Shoes'
  },
  {
    src: rainbowShirt,
    alt: 'Rainbow Shirt',
    title: 'Rainbow Shirt'
  },
  {
    src: womensPants,
    alt: 'Womens Pants',
    title: 'Womens Pants'
  },
  {
    src: womensShirt,
    alt: 'Womens Shirt',
    title: 'Womens Shirt'
  },
  {
    src: bigLamp,
    alt: 'Big Lamp',
    title: 'Big Lamp'
  },
  {
    src: blueChair,
    alt: 'Blue Chair',
    title: 'Blue Chair'
  },
  {
    src: bookshelf,
    alt: 'Bookshelf',
    title: 'Bookshelf'
  },
  {
    src: colander,
    alt: 'Colander',
    title: 'Colander'
  },
  {
    src: comfyChair,
    alt: 'Comfy Chair',
    title: 'Comfy Chair'
  },
  {
    src: couch,
    alt: 'Couch',
    title: 'Couch'
  },
  {
    src: lamp,
    alt: 'Lamp',
    title: 'Lamp'
  },
  {
    src: standingLight,
    alt: 'Standing Light',
    title: 'Standing Light'
  },
  {
    src: vacuum,
    alt: 'Vacuum',
    title: 'Vacuum'
  },
  {
    src: woodenSpoons,
    alt: 'Wooden Spoons',
    title: 'Wooden Spoons'
  },
  {
    src: blackEgg,
    alt: 'Black Egg Outdoor Chair',
    title: 'Black Egg Outdoor Chair'
  },
  {
    src: outdoorLights,
    alt: 'Outdoor Lights',
    title: 'Outdoor Lights'
  },
  {
    src: pathLights,
    alt: 'Path Lights',
    title: 'Path Lights'
  },
  {
    src: patioChairSet,
    alt: 'Patio Chair Set',
    title: 'Patio Chair Set'
  },
  {
    src: patioSet,
    alt: 'Patio Set',
    title: 'Patio Set'
  },
  {
    src: poolChair,
    alt: 'Pool Chair',
    title: 'Pool Chair'
  },
  {
    src: poolSet,
    alt: 'Pool Set',
    title: 'Pool Set'
  },
  {
    src: rotisserie,
    alt: 'Rotisserie',
    title: 'Rotisserie'
  },
  {
    src: smoker,
    alt: 'Smoker',
    title: 'Smoker'
  },
  {
    src: swingChair,
    alt: 'Swing Chair',
    title: 'Swing Chair'
  },
  {
    src: pcSpeaker,
    alt: 'Computer Speakers',
    title: 'Computer Speakers'
  },
  {
    src: cooler,
    alt: 'PC Cooler Fan',
    title: 'PC Cooler Fan'
  },
  {
    src: cryorigCooler,
    alt: 'Cryorig Cooler',
    title: 'Cryorig Cooler'
  },
  {
    src: keyboardMouse,
    alt: 'Keyboard and Mouse',
    title: 'Keyboard and Mouse'
  },
  {
    src: msi3080,
    alt: 'MSI 3080 gpu',
    title: 'MSI 3080 gpu'
  },
  {
    src: redBlackGpu,
    alt: 'Red and Black gpu',
    title: 'Red and Black gpu'
  },
  {
    src: rgbFan,
    alt: 'RGB Fan',
    title: 'RGB Fan'
  },
  {
    src: rgbKeyboard,
    alt: 'RGB Keyboard',
    title: 'RGB Keyboard'
  },
  {
    src: speakerSet,
    alt: 'Speaker Set',
    title: 'Speaker Set'
  },
  {
    src: woodenSpeakers,
    alt: 'Wooden Speakers',
    title: 'Wooden Speakers'
  }
];

export default Images;
