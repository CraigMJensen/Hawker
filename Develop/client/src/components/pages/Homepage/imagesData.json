[
  {
    "src": "../../../assets/images/auto/black_car.jpg",
    "alt": "Black Car",
    "title": "Black Car",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/car_cover.jpg",
    "alt": "Car Cover",
    "title": "Car Cover",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/car_jack.jpg",
    "alt": "Car Jack",
    "title": "Car Jack",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/car_vacuum.jpg",
    "alt": "Car Vacuum",
    "title": "Car Vacuum",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/child-car-seat.jpg",
    "alt": "Child Car Seat",
    "title": "Child Car Seat",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/fuzzy_pink_accesories.jpg",
    "alt": "Fuzzy Pink Accesories",
    "title": "Fuzzy Pink Accesories",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/Jack-Side-by-Side.jpg",
    "alt": "Jacks",
    "title": "Jacks",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/minivan.jpg",
    "alt": "Minivan",
    "title": "Minivan",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/silver_car.jpg",
    "alt": "Silver Car",
    "title": "Silver Car",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/auto/slightly_used.png",
    "alt": "Slightly Used Car",
    "title": "Slightly Used Car",
    "category": "auto"
  },
  {
    "src": "../../../assets/images/clothing/baby_clothes.jpg",
    "alt": "Baby Clothes",
    "title": "Baby Clothes",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/child_pjs.jpg",
    "alt": "Child PJs",
    "title": "Child PJs",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/dress_shoes.jpg",
    "alt": "Dress Shoes",
    "title": "Dress Shoes",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/green_shirt.png",
    "alt": "Green Shirt",
    "title": "Green Shirt",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/mens_pants1.jpg",
    "alt": "Mens Pants",
    "title": "Mens Pants",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/mens_shirt1.jpg",
    "alt": "Mens Shirt",
    "title": "Mens Shirt",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/pizza_shoes.jpg",
    "alt": "Pizza Shoes",
    "title": "Pizza Shoes",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/rainbow_shirt.jpg",
    "alt": "Rainbow Shirt",
    "title": "Rainbow Shirt",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/womens_pants1.jpg",
    "alt": "Womens Pants",
    "title": "Womens Pants",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/clothing/womens_shirt2.jpg",
    "alt": "Womens Shirt",
    "title": "Womens Shirt",
    "category": "clothing"
  },
  {
    "src": "../../../assets/images/household/big_lamp.jpeg",
    "alt": "Big Lamp",
    "title": "Big Lamp",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/blue_chair.jpg",
    "alt": "Blue Chair",
    "title": "Blue Chair",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/bookshelf.jpg",
    "alt": "Bookshelf",
    "title": "Bookshelf",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/colander.jpg",
    "alt": "Colander",
    "title": "Colander",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/comfy_chair.jpeg",
    "alt": "Comfy Chair",
    "title": "Comfy Chair",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/couch.jpg",
    "alt": "Couch",
    "title": "Couch",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/lamp.jpg",
    "alt": "Lamp",
    "title": "Lamp",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/standing_light.jpg",
    "alt": "Standing Light",
    "title": "Standing Light",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/vacuum.jpg",
    "alt": "Vacuum",
    "title": "Vacuum",
    "category": "household"
  },
  {
    "src": "../../../assets/images/household/wooden_spoons.jpg",
    "alt": "Wooden Spoons",
    "title": "Wooden Spoons",
    "category": "household"
  },
  {
    "src": "../../../assets/images/outdoor/acapulco-black-egg-outdoor-chair.jpg",
    "alt": "Black Egg Outdoor Chair",
    "title": "Black Egg Outdoor Chair",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/outdoor_lights.jpg",
    "alt": "Outdoor Lights",
    "title": "Outdoor Lights",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/path_lights.jpg",
    "alt": "Path Lights",
    "title": "Path Lights",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/patio_chair_set.jpeg",
    "alt": "Patio Chair Set",
    "title": "Patio Chair Set",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/patio_set.jpg",
    "alt": "Patio Set",
    "title": "Patio Set",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/pool_chair.jpg",
    "alt": "Pool Chair",
    "title": "Pool Chair",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/pool_set.jpg",
    "alt": "Pool Set",
    "title": "Pool Set",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/rotisserie.jpg",
    "alt": "Rotisserie",
    "title": "Rotisserie",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/smoker.jpg",
    "alt": "Smoker",
    "title": "Smoker",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/outdoor/swing_chair.jpg",
    "alt": "Swing Chair",
    "title": "Swing Chair",
    "category": "outdoor"
  },
  {
    "src": "../../../assets/images/tech/computer-speaker.jpg",
    "alt": "Computer Speakers",
    "title": "Computer Speakers",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/cooler.jpg",
    "alt": "PC Cooler Fan",
    "title": "PC Cooler Fan",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/cryorig-cooler.jpg",
    "alt": "Cryorig Cooler",
    "title": "Cryorig Cooler",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/keyboard_mouse.jpg",
    "alt": "Keyboard and Mouse",
    "title": "Keyboard and Mouse",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/MSI_3080.jpg",
    "alt": "MSI 3080 gpu",
    "title": "MSI 3080 gpu",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/red_black_gpu.jpg",
    "alt": "Red and Black gpu",
    "title": "Red and Black gpu",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/rgb_fan.jpg",
    "alt": "RGB Fan",
    "title": "RGB Fan",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/rgb_keyboard.jpg",
    "alt": "RGB Keyboard",
    "title": "RGB Keyboard",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/speaker_set.jpg",
    "alt": "Speaker Set",
    "title": "Speaker Set",
    "category": "tech"
  },
  {
    "src": "../../../assets/images/tech/wooden_speakers.jpg",
    "alt": "Wooden Speakers",
    "title": "Wooden Speakers",
    "category": "tech"
  }
]
