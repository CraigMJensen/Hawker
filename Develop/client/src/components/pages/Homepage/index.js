import React, { useState} from 'react';
import { useQuery } from '@apollo/client';
import { QUERY_ITEMS } from '../../../utils/queries';
import imagesData from './imagesData.json';
import SingleItemModal from '../SingleItem';
import { Modal, Button } from 'react-bootstrap';

function Homepage() {
  const { data: itemData } = useQuery(QUERY_ITEMS);
  console.info(itemData);

  // getting item data from the database and mapping to the ui
  const getItemData = () => {
    return itemData.item.map((item) => (
      <div className="card" key={item._id}>
        <div>{item.description}</div> 
        <div>{item.price}</div>
        <div>{item.quantity}</div>
        <div>{item.image}</div>
        <div>{item.category.name}</div>
      </div>
    ));
  };

  let item = itemData
  if(itemData){
  return (
    <div>
        {itemData.item.map((item) => {

            {return <div>
                <p>{item.category.name}</p>
                <SingleItemModal props={item}></SingleItemModal>
                </div>}

        })}
    </div>
      
  );
}
}
  

//   return (
//     <div>
//       {imagesData.map((image, i) => (
//         <a href="http://localhost:3000" key={i}>
//           <div className="imageGrid">
//             <img {...imagesData[i]} alt={imagesData[i].alt} width="25%" />
//             <p>{image.category}</p>
//           </div>
//         </a>
//       ))}
//       {itemData ? (getItemData()) : <div>Loading...</div>}
      
//     </div>
//   );
//}

export default Homepage;
