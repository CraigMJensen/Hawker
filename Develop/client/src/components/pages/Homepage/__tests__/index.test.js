// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Homepage from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('Homepage component', () => {
  // verify the component is rendering properly
  it('Homepage component renders', () => {
    render(<Homepage />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Homepage />);
    expect(asFragment()).toMatchSnapshot();
  });
});