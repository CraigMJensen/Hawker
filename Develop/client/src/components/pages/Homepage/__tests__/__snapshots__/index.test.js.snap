// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Homepage
    </h2>
  </div>
</DocumentFragment>
`;
